### PR TITLE
Add one more caching layer to UCRUEADataset

### PR DESCRIPTION
**Pull request by felixdivo**

### CODE REVIEW

File: tests/test_ucr_dataset.py
Comment (mauricekraus) [inline tests/test_ucr_dataset.py:1]: Can you add timing tests to this file?

Comment (felixdivo) [reply tests/test_ucr_dataset.py:1]: I'd rather not, they are nasy to go get running reliably in different environemnts and different versions of numpy etc.

Comment (felixdivo) [reply tests/test_ucr_dataset.py:1]: I tested it manually and the tests are there to ensure nothing raises exceptions

Comment (mauricekraus) [reply tests/test_ucr_dataset.py:1]: Maybe we should add an issue for someone to add it later (maybe the person finds something robust)

**Pull request merged**